### PR TITLE
[tensorflow] [build] [test] TF binaries for 1.15.4 with cu100 py36 and py37

### PR DESCRIPTION
**Pull request by Teng-xu**
*Issue #, if available:*

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Checklist
* When creating a PR:
- [ ] I've modified `src/config/build_config.py` in my PR branch by setting `ENABLE_EI_MODE = True` or `ENABLE_NEURON_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above
#### Benchmark Checklist
* When creating a PR:
- [ ] I've modified `src/config/test_config.py` in my PR branch by setting `ENABLE_BENCHMARK_DEV_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above

## Reviewer Checklist
* For reviewer, before merging, please cross-check:
- [ ] I've verified the code change on the config file mentioned above has already been reverted

*Description:*

*Tests run:*

*DLC image/dockerfile:*

*Additional context:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tensorflow/training/docker/1.15.4/py3/Dockerfile.cpu [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+ENV PYTHONIOENCODING=UTF-8
+ENV LANG=C.UTF-8
+ENV LC_ALL=C.UTF-8
+# Specify the location of module that contains the training logic for SageMaker
+# https://docs.aws.amazon.com/sagemaker/latest/dg/docker-container-environmental-variables-entrypoint.html
+ENV SAGEMAKER_TRAINING_MODULE=sagemaker_tensorflow_container.training:main
+
+# Define framework-related package sources
+ARG TF_URL=https://aws-tensorflow-binaries.s3-us-west-2.amazonaws.com/tensorflow/r1.15_aws/20201014-122550/cpu/py37/tensorflow_cpu-1.15.4-cp37-cp37m-manylinux2010_x86_64.whl
+ARG SMDEBUG_VERSION=0.9.3
```
Comment (saimidu) [inline tensorflow/training/docker/1.15.4/py3/Dockerfile.cpu:27]: Please verify whether SMDEBUG_VERSION needs to be upgraded to 0.9.4

File: tensorflow/training/docker/1.15.4/py3/cu100/Dockerfile.gpu [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+ENV PYTHONIOENCODING=UTF-8
+ENV LANG=C.UTF-8
+ENV LC_ALL=C.UTF-8
+# Specify the location of module that contains the training logic for SageMaker
+# https://docs.aws.amazon.com/sagemaker/latest/dg/docker-container-environmental-variables-entrypoint.html
+ENV SAGEMAKER_TRAINING_MODULE=sagemaker_tensorflow_container.training:main
+
+# Define framework-related package sources
+ARG TF_URL=https://aws-tensorflow-binaries.s3-us-west-2.amazonaws.com/tensorflow/r1.15_aws/20201014-122550/gpu/py37/tensorflow_gpu-1.15.4-cp37-cp37m-manylinux2010_x86_64.whl
+ARG SMDEBUG_VERSION=0.9.3
```
Comment (saimidu) [inline tensorflow/training/docker/1.15.4/py3/cu100/Dockerfile.gpu:24]: Please verify whether SMDEBUG_VERSION needs to be upgraded to 0.9.4

File: tensorflow/training/docker/1.15.4/py36/Dockerfile.cpu [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+# Specify the location of module that contains the training logic for SageMaker
+# https://docs.aws.amazon.com/sagemaker/latest/dg/docker-container-environmental-variables-entrypoint.html
+ENV SAGEMAKER_TRAINING_MODULE=sagemaker_tensorflow_container.training:main
+
+# Define framework-related package sources
+ARG TF_URL=https://aws-tensorflow-binaries.s3-us-west-2.amazonaws.com/tensorflow/r1.15_aws/20201014-122550/cpu/tensorflow_cpu-1.15.4-cp36-cp36m-manylinux2010_x86_64.whl
+
+# The smdebug pipeline relies for following format to perform string replace and trigger DLC pipeline for validating
+# the nightly builds. Therefore, while updating the smdebug version, please ensure that the format is not disturbed.
+ARG SMDEBUG_VERSION=0.9.3
```
Comment (saimidu) [inline tensorflow/training/docker/1.15.4/py36/Dockerfile.cpu:30]: Please verify whether SMDEBUG_VERSION needs to be upgraded to 0.9.4

File: tensorflow/training/docker/1.15.4/py36/cu100/Dockerfile.gpu [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+# Specify the location of module that contains the training logic for SageMaker
+# https://docs.aws.amazon.com/sagemaker/latest/dg/docker-container-environmental-variables-entrypoint.html
+ENV SAGEMAKER_TRAINING_MODULE=sagemaker_tensorflow_container.training:main
+
+# Define framework-related package sources
+ARG TF_URL=https://aws-tensorflow-binaries.s3-us-west-2.amazonaws.com/tensorflow/r1.15_aws/20201014-122550/gpu/tensorflow_gpu-1.15.4-cp36-cp36m-manylinux2010_x86_64.whl
+
+# The smdebug pipeline relies for following format to perform string replace and trigger DLC pipeline for validating
+# the nightly builds. Therefore, while updating the smdebug version, please ensure that the format is not disturbed.
+ARG SMDEBUG_VERSION=0.9.3
```
Comment (saimidu) [inline tensorflow/training/docker/1.15.4/py36/cu100/Dockerfile.gpu:27]: Please verify whether SMDEBUG_VERSION needs to be upgraded to 0.9.4

**Review by saimidu — COMMENTED** [review]
LGTM. Will approve once CI tests succeed.

File: tensorflow/training/docker/1.15.4/py36/cu100/Dockerfile.gpu (hunk 4 above)
Comment (Teng-xu) [reply tensorflow/training/docker/1.15.4/py36/cu100/Dockerfile.gpu]: Verified, updated to 0.9.4

File: tensorflow/training/docker/1.15.4/py36/Dockerfile.cpu (hunk 3 above)
Comment (Teng-xu) [reply tensorflow/training/docker/1.15.4/py36/Dockerfile.cpu]: Verified, updated to 0.9.4

File: tensorflow/training/docker/1.15.4/py3/cu100/Dockerfile.gpu (hunk 2 above)
Comment (Teng-xu) [reply tensorflow/training/docker/1.15.4/py3/cu100/Dockerfile.gpu]: Verified, updated to 0.9.4

File: tensorflow/training/docker/1.15.4/py3/Dockerfile.cpu (hunk 1 above)
Comment (Teng-xu) [reply tensorflow/training/docker/1.15.4/py3/Dockerfile.cpu]: Verified, updated to 0.9.4

**Review by saimidu — APPROVED** [review]

**Pull request merged**